scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/Meta.scala

```diff
@@ -0,0 +1,82 @@
+package scala.meta
+package internal
+package semantic
+
+import java.nio.charset.Charset
+import org.scalameta.data._
+import scala.{Seq => _}
+import scala.collection.immutable.Seq
+import scala.meta.io._
+import scala.meta.internal.semantic.{schema => s}
+import scala.meta.{semantic => m}
+import scala.meta.{Dialect => mDialect}
+import scala.meta.inputs.{Input => mInput, Position => mPosition, Point => mPoint}
+
+package object meta {
+  implicit class XtensionDatabaseSchema(mdatabase: m.Database) {
+    def toSchema(sourcepath: Sourcepath): s.Database = {
+      val sentries = mdatabase.entries.map {
+        case (minput, m.Attributes(mdialect, mnames, mmessages, mdenots, msugars)) =>
+          object mDialect {
+            def unapply(mdialect: mDialect): Option[String] = {
+              val isStandard = scala.meta.Dialect.standards.exists(_._2 == mdialect)
+              if (isStandard) Some(mdialect.toString) else None
+            }
+          }
+          object mPosition {
+            def unapply(mpos: mPosition): Option[s.Range] = mpos match {
+              case scala.meta.inputs.Position.Range(`minput`, mPoint.Offset(_, sstart), mPoint.Offset(_, send)) => Some(s.Range(sstart, send))
```

Comment:
PS, here is how this line looks in the review UI

<img width="1126" alt="screen shot 2017-05-11 at 10 40 45" src="https://cloud.githubusercontent.com/assets/1408093/25940443/702c0356-3636-11e7-93de-2858a80e0d22.png">

Please try to keep lines ~100 chars